tests/test_internals/test_si.py

```diff
@@ -599,6 +602,96 @@ def test_group_set_custom_description(self):
                     assert nwb.electrodes["group"][i].description == "M1 description"
 
 
+class TestAddElectricalSeries(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """Use common recording objects and values."""
+        cls.num_channels = 3
+        cls.durations = [0.100]  # 3000 samples in the recorder
+        cls.test_recording_extractor = generate_recording(num_channels=cls.num_channels, durations=cls.durations)
+
+    def setUp(self):
+        """Start with a fresh NWBFile, ElectrodeTable, and remapped BaseRecordings each time."""
+        self.nwbfile = NWBFile(
+            session_description="session_description1", identifier="file_id1", session_start_time=testing_session_time
+        )
+
+    def test_default_values(self):
+
+        add_electrical_series(recording=self.test_recording_extractor, nwbfile=self.nwbfile, iterator_type=None)
+
+        acquisition_module = self.nwbfile.acquisition
+        assert "ElectricalSeries_raw" in acquisition_module
+        electrical_series = acquisition_module["ElectricalSeries_raw"]
+
+        assert isinstance(electrical_series.data, H5DataIO)
+
+        compression_parameters = electrical_series.data.get_io_params()
+        assert compression_parameters["compression"] == "gzip"
+        assert compression_parameters["compression_opts"] == 4
+
+        extracted_data = electrical_series.data[:]
+        expected_data = self.test_recording_extractor.get_traces(segment_index=0)
+        np.testing.assert_array_equal(expected_data, extracted_data)
+
+    def test_write_as_lfp(self):
+        write_as = "lfp"
+        add_electrical_series(
+            recording=self.test_recording_extractor, nwbfile=self.nwbfile, iterator_type=None, write_as=write_as
+        )
+
+        processing_module = self.nwbfile.processing
+        assert "ecephys" in processing_module
+
+        ecephys_module = processing_module["ecephys"]
+        assert "LFP" in ecephys_module.data_interfaces
+
+        lfp_container = ecephys_module.data_interfaces["LFP"]
+        assert isinstance(lfp_container, pynwb.ecephys.LFP)
+        assert "ElectricalSeries_lfp" in lfp_container.electrical_series
```

Comment:
This sequence of assertions (or analogous for acquisition) are repeated multiple times.

I'd recommend defining your own `TestCase` assertion function such as

```python
def assert_processing_container_properties(
    self,
    nwbfile,
    container_name="LFP",  # for example
    object_name="ElectricalSeries_lfp",  # for example
):
        module = self.nwbfile.processing
        assert "ecephys" in  module

        ecephys_submodule = module["ecephys"]
        assert "LFP" in ecephys_submodule.data_interfaces

        lfp_container = ecephys_submodule.data_interfaces["LFP"]
        assert isinstance(lfp_container, pynwb.ecephys.LFP)
        assert "ElectricalSeries_lfp" in lfp_container.electrical_series
```

which would not only reduce the total code by a bit but also make those tests more unit-like by indicating in one line what behavior is being asserted (even if the way that it is asserted involves multiple aspects like this).

Reply:
I think that is a good idea but there are really only two cases as the acquisition boilerplate has a simpler structure. Plus, the `isinstance` for the container type is different in each of the two cases so that's an extra parameter.  I feel that introducing indirection in a test to cover only two cases gains in code reduction what it loses in code clarity (even if intent clarity becomes better). What do you think? I am on the edge.

Reply:
Depends on if that kind of assertion could be used in any other tests through the suite? Or for more general objects (not just ElectricalSeries)

Such an extension of a `TestCase` could be defined in the testing module for usage across the suite.

Up to you, though (can do it in subsequent PR if you decide yes)

Reply:
Yes, that sounds like a good idea. I will check it out once I add more tests, maybe the pattern will repeat itself more.